__mocks__/fs.js

```diff
@@ -0,0 +1,21 @@
+const fs = jest.genMockFromModule('fs')
```

Comment:
[Inspired by the Jest documentation](https://jestjs.io/docs/en/manual-mocks), this `fs` mock is generated by the Jest framework, but I've overridden a couple things. Let me know if this makes sense:

* Call through the real filesystem when calling `fs.readdir` and `fs.readFile` so that we allow our CLI-under-test to access the real template it needs to bootstrap a GitHub action directory
* Overrwrite `fs.writeFile` with a custom implementation that just writes the file path and its contents to memory (I chose a Map, but could also just use a plain object).

Reply:
Sounds good - appreciate the detailed explanation 🙇 `Map` makes sense here 👍